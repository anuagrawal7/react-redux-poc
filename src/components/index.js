import Header from './header/header';
import Footer from './footer';
import UsersList from './userList/userList';
import UserUpdateForm from './userUpdateForm';

export {
  Header,
  Footer,
  UsersList,
  UserUpdateForm
}