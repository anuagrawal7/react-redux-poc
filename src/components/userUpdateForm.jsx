import React, { useMemo, useCallback } from 'react';
import { useForm } from './../hooks/useForm';
import { Button, Input } from './../shared';

const UserUpdateForm = ({ initialValues, onSave }) => {
  const manipulateData = useMemo(() => {
    let defaultData = {};

    initialValues.forEach(ele => {
      defaultData[ele['field']] = ele['value'];
    });
    return defaultData;
  }, [initialValues]);

  const updateData = useCallback((formData, formErr = {}) => {
    const errs = Object.keys(formErr);
    if (errs && !errs.length) {
      onSave(formData)
    }
  }, [onSave]);

  const { values, onChange, reset, onBlur, error } = useForm(manipulateData);

  return (
    <section>
      <form className="form-group">
        {initialValues.map((ele, index) => {
          return <Input key={index}
            name={ele['field']}
            value={values[ele['field']]}
            onChange={onChange}
            type={ele['type']}
            onBlur={onBlur}
            label={ele['label']}
            validations={ele.validations} error={error} />
        })}
        <Button onClick={() => updateData(values, error)} classList='btn-primary margin-ry-10' >Save</Button>
        <Button onClick={() => reset(manipulateData)} classList='btn-secondary' >Reset</Button>

      </form>
    </section>
  )
}

export default React.memo(UserUpdateForm);