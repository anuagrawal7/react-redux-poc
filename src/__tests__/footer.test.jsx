
import React from 'react';
import { shallow } from 'enzyme';
import { Footer } from './../components/';

describe("Footer component", () => {
  it("should not crash", () => {
    const wrapper = shallow(<Footer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})