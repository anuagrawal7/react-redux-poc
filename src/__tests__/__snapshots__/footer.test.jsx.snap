// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should not crash 1`] = `"{}"`;
